packages/origin-ui-core/src/utils/exchange/math.ts

```diff
@@ -8,3 +8,34 @@ export function calculateTotalPrice(priceInDisplayUnit: string, energyInDisplayU
 
     return (priceAsFloat * energyAsFloat).toFixed(2);
 }
+
+export function calculateDemandTotalVolume(period, volume, startDate, endDate) {
+    if (period && volume && startDate && endDate) {
+        const start = +new Date(startDate);
+        const end = +new Date(endDate);
+        const dayDifference = Math.round((end - start) / (60 * 60 * 24 * 1000));
+        const parsedVolume = parseInt(volume, 10);
+        const setDivisor = (timeframe) => {
+            switch (timeframe) {
+                case 0:
+                    return 365;
+                case 1:
+                    return 30;
+                case 2:
+                    return 1;
+                case 3:
+                    return 7;
+                default:
+                    return 1;
+            }
+        };
+        const divisor = setDivisor(period);
+
+        if (dayDifference > 0 && parsedVolume > 0) {
+            const numberOfOccurrences = Math.round(dayDifference / divisor) + 1;
+            return parsedVolume * numberOfOccurrences;
+        }
+    } else {
+        return '';
+    }
+}
```

Comment:
This potentially can lead to a situation where UI would present different outcome than it would be then created the API. We should basically use same code/logic for generation on the UI side.

I suggest we add new endpoint such as to `POST /demand/prepare` that would return a populated demand object with bids etc that can be verified by the user before creating a real demand.

Reply:
Ok, will adjust code according to comments